lib/galaxy/jobs/runners/util/file-staging.py

```diff
@@ -0,0 +1,41 @@
+import os
+import sys
+
+from pulsar.client.transport import post_file
+
+
+def get_job_directory_files(work_dir: str):
+    """
+    Get path for all the files from work directory
+    """
+    paths = []
+    for root, _, files in os.walk(work_dir):
+        for file in files:
+            paths.append(os.path.join(root, file))
+
+    return paths
+
+
+def stage_back_data(url, paths):
+    """Responsible for staging out of data for the desired path"""
+
+    url = url.replace("localhost", str("172.17.0.1"))
+    url = url.replace("127.0.0.1", str("172.17.0.1"))
```

Comment:
This looks odd, do you need to replace this with galaxy_infrastructure_url ?

Reply:
Actually, galaxy_infrastructure_url is also used for generating links for all the upload files.  This script runs inside a docker container, so for it to handle and reroute the requests in case of localhost, it needs to replace the URL with the default docker address. You can find something similar in the test_kubernetes_staging.py file. There it replaces with a similar host pointing to the same address. If you want, I can replace it similarly

Reply:
You cannot rely on 172.17.0.1, this only works for the default bridge network that docker sets up. That is why the kubernetes staging test exposes this as a variable. Note also that this is just for testing purposes and is using the galaxy_infrastructure_url mechansim

Reply:
Should I remove this? Although it will replace only in localhost cases

Reply:
Can you make this configurable ?

Reply:
Configurable as in? The IP address would be specified in the configs and use, or replace with `172.17.0.1` only for testing purposes?

Reply:
We generally do not hardcode things that are only needed for tests. Is this the url for the host that the TES service runs on or is it the URL galaxy runs on ? In both cases this should exclusively be configured via the configuration. If that is the case you can remove the IP replacement you're doing here.

Reply:
it is the URL that galaxy runs on, but the issue is the docker container inside TES cannot communicate directly with this network, but TES can, so during the building of the config file for job, the galaxy URL is used directly for getting the URL of input files, but the same URL cannot be used for sending back because of docker container being on a different network

Reply:
ok, I can fix it away, if the staging out URL is specified specifically I will use that link, by default I will continue using the same link as Galaxy

Reply:
Fixed